apps/tlon-mobile/src/utils/branch.ts

```diff
@@ -0,0 +1,99 @@
+import * as urbit from '@tloncorp/shared/dist/urbit';
+import { isValidPatp } from 'urbit-ob';
+
+import { BRANCH_DOMAIN, BRANCH_KEY } from '../constants';
+
+const fetchBranchApi = async (path: string, init?: RequestInit) =>
+  fetch(`https://api2.branch.io${path}`, init);
+
+export const getDeepLink = async (alias: string) => {
+  const params = new URLSearchParams();
+  params.set('url', `https://${BRANCH_DOMAIN}/${alias}`);
+  params.set('branch_key', BRANCH_KEY);
+  const response = await fetchBranchApi(`/v1/url?${params}`);
+
+  console.log(`/v1/url?${params}`);
+
+  if (!response.ok) {
+    return undefined;
+  }
+
+  const {
+    data: { url },
+  } = (await response.json()) as {
+    data: { url: string };
+  };
+  return url;
+};
+
+export type DeepLinkType = 'lure' | 'wer';
+interface DeepLinkData {
+  $desktop_url: string;
+  $canonical_url: string;
+  lure?: string;
+  wer?: string;
+}
+
+export async function getDmLink(ship: string): Promise<string> {
+  const dmPath = `dm/${ship}`;
+  const fallbackUrl = `https://tlon.network/lure/~loshut-lonreg/tlon`; // for now, send to generic signup page on desktop
+  const link = await createDeepLink(fallbackUrl, 'wer', dmPath);
+  return link || '';
+}
+
+export const createDeepLink = async (
+  fallbackUrl: string | undefined,
+  type: DeepLinkType,
+  path: string
+) => {
+  if (!fallbackUrl || !path) {
+    return undefined;
+  }
+
+  if (type === 'lure' && !urbit.whomIsFlag(path)) {
+    return undefined;
+  }
+
+  if (type === 'wer') {
+    const [location, ship] = path.split('/');
```

Comment:
I think we could use `api.getPostInfoFromWer` for this.

Reply:
Well, that's handy :)